networking_ccloud/ml2/agent/eos/switch.py

```diff
@@ -236,6 +242,72 @@ def get_removable_translations_cmds(iface_name, wanted_translations):
 
         return commands
 
+    def get_config(self):
+        # get infos from the device for everything that we have a model for
+        config = agent_msg.SwitchConfigUpdate(switch_name=self.name, operation=Op.add)
+
+        # vlans
+        for vlan, data in self.send_cmd("show vlan")['vlans'].items():
+            config.add_vlan(int(vlan), data['name'])
+
+        # vxlan_maps
+        vxlan_maps = self.send_cmd("show interfaces Vxlan1")['interfaces']['Vxlan1']['vlanToVniMap']
+        for vlan, data in vxlan_maps.items():
+            if 'vni' in data:
+                config.add_vxlan_map(int(data['vni']), int(vlan))
+
+        # bgp
+        bgp_sum = self.send_cmd("show bgp summary")['vrfs']['default']
+        config.bgp = agent_msg.BGP(asn=bgp_sum['asn'], asn_region=self.drv_conf.global_config.asn_region, vlans=[])
+        curr_bgp_vlans = self.send_cmd("show bgp evpn instance")['bgpEvpnInstances']
+        vre = re.compile(r"VLAN (?P<vlan>\d+)")
+        for vlan_name, data in curr_bgp_vlans.items():
+            m = vre.match(vlan_name)
+            if not m:
+                LOG.warning("Could not match bgp vpn instanec name '%s'", vlan_name)
+                continue
+
+            bv = agent_msg.BGPVlan(rd=data['rd'], vlan=int(m.group('vlan')),
+                                   rt_imports=data.get('importRts', []), rt_exports=data.get('exportRts', []),
+                                   rd_evpn_domain_all='remoteRd' in data,
+                                   rt_imports_evpn=data.get('importRemoteRts', []),
+                                   rt_exports_evpn=data.get('exportRemoteRts', []))
+
+            config.bgp.vlans.append(bv)
+            # FIXME: redistribute learned flag?
+
+        # ifaces - vlans
+        ifaces_vlans = self.send_cmd("show interfaces vlans")['interfaces']
+        for name, data in ifaces_vlans.items():
+            if name == "Vxlan1":
+                continue
+            iface = config.get_or_create_iface(name)
+            if data.get("untaggedVlan", 1) != 1:
```

Comment:
Can this be `'1'`? We converted a lot to `int()` above, so I'm curious why this comes back as `int` already.

Reply:
Values in the `EOS` json response are mostly an int where they should be an int. Dict keys on the other hand are strings, so they need int converting.

```
qa-de-1-sw4113a-bb271#show interfaces vlans | json | inc untagged
            "untaggedVlan": 2121
            "untaggedVlan": 2121
            "untaggedVlan": 2121
```